src/librustc/hir/check_attr.rs

```diff
@@ -53,34 +53,39 @@ impl<'a> CheckAttrVisitor<'a> {
             }
         };
         for word in words {
-            let word: &str = &word.name();
-            let message = match word {
+            let name = match word.name() {
+                Some(word) => word,
+                None => continue,
+            };
+
+            let message = match &*name {
                 "C" => {
                     if target != Target::Struct && target != Target::Enum {
-                            "attribute should be applied to struct or enum"
+                        "attribute should be applied to struct or enum"
                     } else {
                         continue
                     }
                 }
                 "packed" |
-                "simd" => {
-                    if target != Target::Struct {
-                        "attribute should be applied to struct"
-                    } else {
-                        continue
+                    "simd" => {
+                        if target != Target::Struct {
+                            "attribute should be applied to struct"
+                        } else {
+                            continue
+                        }
```

Comment:
The arm would look nicer (imo) and have less rightward drift as
```rust
                "packed" | "simd" => {
                    if target != Target::Struct {
                        "attribute should be applied to struct"
                    } else {
                        continue
                    }
                }
```
than
```rust
                "packed" |
                    "simd" => {
                        if target != Target::Struct {
                           "attribute should be applied to struct"
                        } else {
                            continue
                        }
                    }
```
I haven't seen the style with the extra indent anywhere in the compiler before, but I suppose `rustfmt` recently adopted it...

Regardless, orthogonal formatting changes make reviewing more difficult -- for future reference, perhaps do a separate commit for `rustfmt` changes.